docs/rules/require-migration-script-metadata.md

```
@@ -0,0 +1,117 @@
+
+This rule enforces a single JSDoc block at the top of the file (before any imports or statements) that contains the following tags:
+
+- `@migration`: `true` or `false` (Required)
+- If `@migration true`, the following are also required:
+  - `@migrationPhase`: `before` or `after`
+  - `@migrationDependencies`: `NONE` or a comma-separated list of script names (without `.f.ts`)
+  - `@migrationDescription`: A brief description of what the script does
+
+### Examples of Incorrect Code
+
+#### Missing metadata
```

markdownlint-cli2 (0.18.1)

Headings should be surrounded by blank lines
Expected: 1; Actual: 0; Below

(MD022, blanks-around-headings)